packages/block-editor/src/components/list-view/branch.js

```diff
@@ -116,6 +117,16 @@ function ListViewBranch( props ) {
 
 	const { expandedState, draggedClientIds } = useListViewContext();
 
+	// For long lists where the selected item may fall outside of the current window,
+	// pass a reference to the corresponding placeholder row for the selected item.
+	// The "real" selected item is also observed in ListViewBlock, when rendered.
+	const invisibleSelectedItemRef = useRef();
+	useListViewScrollIntoView( {
+		firstSelectedBlockClientId: selectedClientIds?.[ 0 ],
+		numBlocksSelected: selectedClientIds?.length,
```

Comment:
The interface could potentially be simplified to just `useListViewScrollIntoView( { selectedClientIds, selectedItemRef } )`, moving the logic that grabs the first clientId and the length into the hook.